Source/Core/DolphinQt2/Config/Mapping/MappingButton.cpp

```diff
@@ -138,25 +124,38 @@ bool MappingButton::event(QEvent* event)
 
 void MappingButton::mouseReleaseEvent(QMouseEvent* event)
 {
-  if (m_reference->IsInput())
+  switch (event->button())
   {
-    switch (event->button())
+  case Qt::MouseButton::LeftButton:
+    if (m_parent->GetParent()->GetDevice() == nullptr)
+      return;
+    if (m_reference->IsInput())
     {
-    case Qt::MouseButton::LeftButton:
       QPushButton::mouseReleaseEvent(event);
-      break;
-    case Qt::MouseButton::MiddleButton:
-      Clear();
-      break;
-    case Qt::MouseButton::RightButton:
-      // TODO Open advanced dialog
-      break;
-    default:
-      break;
     }
+    else
+    {
+      m_io_window->Update();
+      m_io_window->exec();
+      Update();
+    }
+    return;
+  case Qt::MouseButton::MiddleButton:
+    Clear();
+    break;
+  case Qt::MouseButton::RightButton:
+    if (m_parent->GetParent()->GetDevice() == nullptr)
```

Comment:
Is this related to the suggestion above? If it is, yes that would only be logical then.